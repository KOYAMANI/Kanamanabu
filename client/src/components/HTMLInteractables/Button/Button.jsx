import React from "react";
import styles from "./Button.module.css";

function Button(props) {
  return <button className={styles.Button} type={props.type}>{props.text}</button>;
}

export default Button;
