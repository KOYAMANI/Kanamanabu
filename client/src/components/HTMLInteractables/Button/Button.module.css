* {
    box-sizing: border-box;
}

.button {
    --height: 3rem;
    --mainColor: black  ;
    --secondaryColor: white;


    height: var(--height);

    color: var(--secondaryColor);
    border: 3px solid var(--secondaryColor);
    background: var(--mainColor);

    display: flex;
    align-items: center;
    justify-content: center;

    cursor: pointer;

    border-radius: 3px;

    font-size: calc(var(--height) / 3);
}
.button.short {
    padding: 0 3rem;
}
.button.long {
    height: 100%;
}
.button.pill {
    border-radius: calc(var(--height) / 2);
}