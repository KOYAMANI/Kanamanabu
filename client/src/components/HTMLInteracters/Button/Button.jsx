import React from "react";

function Button(props) {
  return <button className="w-full border-box m-1 bg-gray-800 px-6 py-2.5 text-white rounded-sm hover:bg-gray-700 " type={props.type}>{props.text}</button>;
}

export default Button;
