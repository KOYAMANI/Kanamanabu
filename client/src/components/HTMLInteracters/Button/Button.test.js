import {render, screen, cleanup} from '@testing-library/react';
import ReactDOM from "react-dom";
import renderer from 'react-test-renderer';
import Button from "./Button";

afterEach(cleanup);

describe('Button', () =>{
    test('Render without crashing', ()=>{
        const div = document.createElement('div')
        ReactDOM.render(<Button text="button-test" type="submit"/>, div)
    });

    test('Render text', ()=>{
        render(<Button text="button-test" type="submit"/>)
        const element = screen.getByText('button-test')
        expect(element).toBeInTheDocument();
    });

    test('match snapshot',()=>{
        const tree = renderer.create(<Button text="button-test" type="submit"/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
})