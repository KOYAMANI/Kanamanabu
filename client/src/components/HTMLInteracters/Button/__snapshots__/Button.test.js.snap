// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button match snapshot 1`] = `
<button
  className="w-full border-box m-1 bg-gray-800 px-6 py-2.5 text-white rounded-sm hover:bg-gray-700 "
  type="submit"
>
  button-test
</button>
`;
